aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -56,6 +56,9 @@ namespace {
 // TODO(eqy): more benchmarking to determine whether this should include sm86/89
 // Needs to be kept in-sync with test_fused_chocie in test_transformers.py
 bool check_prefer_cudnn_attention() {
+  // TODO(eqy): Re-enable by default after upgrading to a release later than 9.5.0
+  // see context: https://github.com/pytorch/pytorch/issues/138340
+  return false;
```

Comment:
I think it okay to keep on if the input query is contiguous

Reply:
Why do we need a release that's later than 9.5.0 for this?

Reply:
Ok, so sound like in terms of 2.5.0, we just want to chery-pick this change?  In that case don't you want to add this logic on line 65 (i.e. CUDNN_VERSION >= 90500)